cajita/src/Cajita_GlobalGrid.hpp

```diff
@@ -157,11 +157,22 @@ class GlobalGrid
     //! \param dim Spatial dimension.
     int ownedNumCell( const int dim ) const;
 
+    //! \brief Set the owned number of cells in a given dimension of this block.
+    //! \param dim Spatial dimension.
+    //! \param num_cell New number of owned cells.
+    void setOwnedNumCell( const int dim, const int num_cell );
```

Comment:
I want this function to take all dimensions at once: `void setOwnedNumCell( const std::array<int,num_space_dim>& num_cell )`. In addition, I want a private function `computeGlobalOffset()` that automatically computes the global offsets given the owned number of cells. That way the partitioners don't need to compute that. This global offset compute function could also be called in the constructor then.

Reply:
Should I also add the corresponding `ownedNumCell` that returns the array?

Reply:
Regarding `computeGlobalOffset`: Are there already other helper functions present or should I just directly call `MPI_Exscan`?

Reply:
With the added `computeGlobalOffset` I assume you also want that automatically called after setting the number of cells, so that everything is always up to date? Which would incur an additional (and for my load balancing case unnecessary) MPI communication.

Reply:
yes - any time the number of cells is changed that will need to be called. it does incur communication but without it other objects that use this class will access it and it is not in a correct parallel consistent state.

Reply:
As discussed this has changed to `setNumCellAndOffset` to force the update to either the number of cells or the offset to coincide. With that writing `computeGlobalOffset` is postponed.